File: python/programmers/hash/전화번호 목록.py
def solution(phone_book):
    answer = True
    p_dict = {}
    for num in phone_book:
        if num not in p_dict:
            p_dict[num] = 1
    for num in phone_book:
        temp = ''
        for index in range(len(num)-1):
            temp += num[index];
            if temp in p_dict:
                return False
    return answer

File: java/src/main/java/ps_traning/programmers/level_2/전화번호_목록.java
package ps_traning.programmers.level_2;

import java.util.HashSet;
import java.util.Set;

public class 전화번호_목록 {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Set<String> set = new HashSet<>();
        for (String num : phone_book) {
            set.add(num);
        }
        for (String num : phone_book) {
            for (int i = 1; i < num.length(); i++) {
                if (set.contains(num.substring(0, i))) {
                    return false;
                }
            }
        }
        return answer;
    }
}
